examples/openvino/stable-diffusion/train_text_to_image_qat.py

```diff
@@ -1,1092 +0,0 @@
-
-                # Predict the noise residual and compute loss
-                noise_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
-
-                loss = F.mse_loss(noise_pred.float(), noise.float(), reduction="mean")
-
-                if args.use_kd:
-                    with torch.no_grad():
-                        orig_output = teacher_model(noisy_latents, timesteps, encoder_hidden_states).sample
-                    loss += F.mse_loss(noise_pred.float(), orig_output.float(), reduction="mean")
-
-                # Gather the losses across all processes for logging (if we use distributed training).
-                avg_loss = accelerator.gather(loss.repeat(args.train_batch_size)).mean()
-                train_loss += avg_loss.item() / args.gradient_accumulation_steps
-
-                compression_loss_unet = compression_controller.loss()
-                loss = loss + compression_loss_unet
-
-                # Backpropagate
-                accelerator.backward(loss)
-                if accelerator.sync_gradients:
-                    accelerator.clip_grad_norm_(unet.parameters(), args.max_grad_norm)
-                optimizer.step()
-                lr_scheduler.step()
-                optimizer.zero_grad()
-
-            # Checks if the accelerator has performed an optimization step behind the scenes
-            if accelerator.sync_gradients:
-                if args.ema_device:
-                    ema_unet.step(orig_unet.parameters())
-                progress_bar.update(1)
-                global_step += 1
-                accelerator.log({"train_loss": train_loss}, step=global_step)
-                train_loss = 0.0
-
-            logs = {"step_loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]}
-            progress_bar.set_postfix(**logs)
-
-            if global_step >= args.max_train_steps:
-                break
-
-    # Create the pipeline using the trained modules and save it.
-    accelerator.wait_for_everyone()
-    if accelerator.is_main_process:
-        unet = accelerator.unwrap_model(unet)
-        if args.ema_device:
-            ema_unet.copy_to(orig_unet.parameters())
-
-        if args.push_to_hub:
-            repo.push_to_hub(commit_message="End of training", blocking=False, auto_lfs_prune=True)
-
-    accelerator.end_training()
-
-    # Export optimized pipline to OpenVINO
-    export_unet = compression_controller.strip(do_copy=False)
-    export_pipeline = StableDiffusionPipeline(
-        text_encoder=text_encoder,
-        vae=vae,
-        unet=export_unet,
-        tokenizer=tokenizer,
-        scheduler=noise_scheduler,
-        safety_checker=pipeline.safety_checker,
-        feature_extractor=pipeline.feature_extractor,
-    )
-
-    save_directory = Path(args.output_dir) / "openvino"
-    export_from_model(export_pipeline, output=save_directory, task="stable-diffusion")
-
-
-if __name__ == "__main__":
-    main()
```

Comment:
could you rename the notebook to `stable_diffusion_hybrid_quantization.ipynb`?

Reply:
also could you add it (+ small description in https://github.com/huggingface/optimum-intel/tree/main/notebooks/openvino)